src/plugins/data/server/search/opensearch_search/decide_client.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchClient, RequestHandlerContext } from 'src/core/server';
+import { IOpenSearchSearchRequest } from '..';
+
+export const decideClient = async (
+  context: RequestHandlerContext,
+  request: IOpenSearchSearchRequest
+): Promise<OpenSearchClient> => {
+  // if data source feature is disabled, return default opensearch client of current user
+  const client =
+    request.dataSourceId && context.dataSource
```

Comment:
This condition implies that the index pattern must be attached to a data source and have the data source feature enabled.

Reply:
if data source is not enabled. `context.dataSource` will be undefined